lib/fluent/plugin/out_statsd.rb

```diff
@@ -60,6 +60,7 @@ def write(chunk)
           send_to_statsd(*metric.values_at(*arg_names).map {|str| parser.parse(str) })
         end
       end
+      @statsd.flush
```

Comment:
I think a better place for this would be within the shutdown method to ensure all metrics have been flushed

Reply:
I should add `flush` call in `shutdown`, so I add it.

But there is a reason why I add `flush` call in `write`.
I want to flush immediately when new metrics input coming.
Because the statsd doesn't have timer to flush, if input rate is very low some input could be leaved in buffer.
The behavior cause problem about correctness of metrics.

https://github.com/reinh/statsd/blob/v1.4.0/lib/statsd.rb#L356
https://github.com/reinh/statsd/blob/v1.4.0/lib/statsd.rb#L488
https://github.com/reinh/statsd/blob/v1.4.0/lib/statsd.rb#L88-L104

Please tell me your opinion about this problem.

Reply:
Sounds like a good workaround, maybe some timer based flushing could be added to the upstream gem to solve this without affecting the bulk size for in all cases

Reply:
Maybe we can add a flexible counter, first, we let it flush every time and record a timestamp, add the flush interval if it is very frequent(such as a minute flush 60 times), but the buffer cannot over `batch_size`. and reduce the interval when the input rate is very low. And I think it should be used in conjunction with time limit to avoid the special situation.
Then, I think it can be a good deal with large flow, and can be timely flush if input rate is very low.